cmd/reviewdog/main.go

```diff
@@ -531,13 +531,23 @@ func githubClient(ctx context.Context, token string) (*github.Client, error) {
 const defaultGitHubAPI = "https://api.github.com/"
 
 func githubBaseURL() (*url.URL, error) {
-	baseURL := os.Getenv("GITHUB_API")
-	if baseURL == "" {
-		baseURL = defaultGitHubAPI
+	if baseURL := os.Getenv("GITHUB_API"); baseURL != "" {
+		u, err := url.Parse(baseURL)
+		if err != nil {
+			return nil, fmt.Errorf("GitHub base URL from GITHUB_API is invalid: %v, %w", baseURL, err)
+		}
+		return u, nil
 	}
-	u, err := url.Parse(baseURL)
+	if baseURL := os.Getenv("GITHUB_API_URL"); baseURL != "" {
+		u, err := url.Parse(baseURL)
+		if err != nil {
+			return nil, fmt.Errorf("GitHub base URL from GITHUB_API_URL is invalid: %v, %w", baseURL, err)
+		}
+		return u, nil
+	}
+	u, err := url.Parse(defaultGitHubAPI)
 	if err != nil {
-		return nil, fmt.Errorf("GitHub base URL is invalid: %v, %w", baseURL, err)
+		return nil, fmt.Errorf("GitHub base URL from reviewdog default is invalid: %v, %w", defaultGitHubAPI, err)
 	}
 	return u, nil
```

Comment:
Something to keep in mind: GitHub Actions does indeed have the GitHub API URL set in `GITHUB_API_URL`, but without the trailing slash, as per [the docs](https://docs.github.com/en/actions/reference/environment-variables): `https://api.github.com`.

In https://github.com/reviewdog/action-tflint/pull/38#issuecomment-845121149 we saw that reviewdog fails when the URL does not end with a trailing slash: 

```
reviewdog: post failed for tflint: fail to parse diff: BaseURL must have a trailing slash, but "https://api.github.com" does not
```

Unless there is one already, it might make sense to add a `/` before returning here. I'll defer to @haya14busa as he's more familiar with the codebase.

Reply:
I think BaseURL has a trailing slash or not is important to make requests, so I think it would be good to remove that restriction.

Reply:
Thanks for notifying me of that point.

Reply:
hmm, the test fails with "BaseURL must have a trailing slash, but "https://api.github.com" does not"

https://github.com/reviewdog/reviewdog/pull/976/checks?check_run_id=2913610699#step:6:7

Reply:
oops. I thought it is a reviewdog's assertion, but it is go-github's.
I'm going to put a slash at the end of the environment variable.

Reply:
I misunderstood that BaseURL has a trailing slash or not is important.
But the result of reference resolution defined in RFC3986 will change whether BaseURL has a trailing slash or not.
So, go-github's assertion is necessary.
https://www.rfc-editor.org/rfc/rfc3986.html#section-5.2.3